src/image_processing/computed_field_derivative_image_filter.cpp

```diff
@@ -155,8 +155,8 @@ Returns allocated command string for reproducing field. Includes type.
 			append_string(&command_string, field_name, &error);
 			DEALLOCATE(field_name);
 		}
-		sprintf(temp_string, " order %d", order);
-		sprintf(temp_string, " direction %d", direction);
+        snprintf(temp_string, 40, " order %d", order);

```

Comment:
Surely must be an append_string between these 2 lines otherwise first snprintf output not used.